src/Npgsql/Schema/DbColumnSchemaGenerator.cs

```diff
@@ -116,16 +119,21 @@ internal ReadOnlyCollection<NpgsqlDbColumn> GetColumnSchema()
                     ? GenerateOldColumnsQuery(columnFieldFilter)
                     : GenerateColumnsQuery(_connection.PostgreSqlVersion, columnFieldFilter);
 
-                using var scope = new TransactionScope(TransactionScopeOption.Suppress);
+                using var scope = new TransactionScope(
+                    TransactionScopeOption.Suppress,
+                    async ? TransactionScopeAsyncFlowOption.Enabled : TransactionScopeAsyncFlowOption.Suppress);
                 using var connection = (NpgsqlConnection)((ICloneable)_connection).Clone();
 
-                connection.Open();
+                if (async)
```

Comment:
There's an Open(bool)

Reply:
I had to make it `internal` as it was `private`